ekncontent/ekncontent/eknc-utils.c

```diff
@@ -370,6 +379,96 @@ eknc_get_data_dir (const gchar *app_id)
   return NULL;
 }
 
+static GList *
+databases_dirs_from_metadata (const gchar *flatpak_path, const gchar *app_id)
+{
+  GList *list = NULL;
+  g_autoptr(GKeyFile) metakey = NULL;
+  g_auto(GStrv) groups = NULL;
+  g_autofree gchar *metadata_path = NULL;
+  g_autofree gchar *runtime = NULL;
+  g_auto(GStrv) components = NULL;
+  gchar *arch = NULL;
+
+  metadata_path = g_build_filename (flatpak_path, "flatpak",
+                                    "app", app_id,
+                                    "current", "active", "metadata",
+                                    NULL);
+  metakey = g_key_file_new ();
+  if (!g_key_file_load_from_file (metakey, metadata_path, G_KEY_FILE_NONE, NULL))
+    return NULL;
+
+  runtime = g_key_file_get_string (metakey, "Application", "runtime", NULL);
+  components = g_strsplit (runtime, "/", 3);
+  arch = components[1];
+
+  groups = g_key_file_get_groups (metakey, NULL);
+  for (int i = 0; groups[i] != NULL; i++)
+    {
+      g_autoptr(GFile) extension_dir = NULL;
+      gchar *extension_name = NULL;
+      g_autofree gchar *extension_version = NULL;
+      g_autofree gchar *extension_path = NULL;
+
+      if (!g_str_has_prefix (groups[i], FLATPAK_EXTENSION_GROUP))
+        continue;
+
+      extension_name = (extension_name = (groups[i] + strlen (FLATPAK_EXTENSION_GROUP)));
+      if (!g_str_has_prefix (extension_name, app_id))
```

Comment:
You could use g_strstr_len() directly in the previous check instead of this strlen magic :)